pkg/kev/config/extensions.go

```diff
@@ -0,0 +1,143 @@
+package config
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/go-playground/validator/v10"
+	"github.com/imdario/mergo"
+	"gopkg.in/yaml.v3"
+)
+
+type ExtensionRoot struct {
+	K8S K8SServiceConfig `yaml:"x-k8s"`
+}
+
+type K8SServiceConfig struct {
+	Enabled   bool      `yaml:"enabled"`
+	Workload  Workload  `yaml:"workload" validate:"required"`
+	Component Component `yaml:"component"`
```

Comment:
`Component` is basically redundant here as we already have `Enabled` field above covering for what we currently support at a [component level](https://github.com/appvia/kev/blob/master/docs/reference/config-params.md#kevcomponentenabled).

Also, [K8s service part](https://github.com/appvia/kev/blob/master/docs/reference/config-params.md#-service) of configuration isn't cover yet, which is fine, but maybe drop it in there as a placeholder `Service` field with a todo action?